### PR TITLE
[clang] CTAD alias: Emit a more descriptive diagnostic message when is_deducible  constraint is evaluated to false.

### LINKED ISSUE #92225: [clang] Improve is_deducible diagnostic for CTAD alias.
With #89358, when the is_deducible constraint is not satisfied for a CTAD alias deduction guide, clang emits the diagnostic note: `because '__is_deducible(BFoo, Foo<int *>)' evaluated to false`. The diagnostic mentions an internal-only type-trait.

Per the discussion, we should emit a more user-friendly and descriptive diagnostic for that, something like: `can not deduce template argument for BFoo from 'Foo<int *>`

### PR DESCRIPTION
**Pull request by hokein**
Fixes https://github.com/llvm/llvm-project/issues/92225

### CODE REVIEW

File: clang/test/SemaCXX/cxx20-ctad-type-alias.cpp [hunk 1]
```diff
@@ -111,7 +111,7 @@ struct Foo {
 template <typename X, int Y>
 using Bar = Foo<X, sizeof(X)>; // expected-note {{candidate template ignored: couldn't infer template argument 'X'}} \
                                // expected-note {{candidate template ignored: constraints not satisfied [with X = int]}} \
-                               // expected-note {{because '__is_deducible}}
+                               // expected-note {{can not deduce template arguments for 'Bar' from the type 'Foo<int, 4UL>'}}
```
[the remark this answers is not in the document]
Comment (hokein) [reply clang/test/SemaCXX/cxx20-ctad-type-alias.cpp]: ah, this was left when I adjusted the message. Done.

**Review by cor3ntin — APPROVED** [review]
LGTM

**Comment by zeroomega** [comment]
Hi @hokein , your change probably broke the test "Clang :: SemaCXX/cxx20-ctad-type-alias.cpp" on Windows. Could you take a look please?

We are seeing test failures on this with error message:

```
Exit Code: 1

Command Output (stdout):
--
# RUN: at line 1
c:\b\s\w\ir\x\w\llvm_build\bin\clang.exe -cc1 -internal-isystem C:\b\s\w\ir\x\w\llvm_build\lib\clang\19\include -nostdsysteminc -fsyntax-only -Wno-c++11-narrowing -Wno-literal-conversion -std=c++20 -verify C:\b\s\w\ir\x\w\llvm-llvm-project\clang\test\SemaCXX\cxx20-ctad-type-alias.cpp
# executed command: 'c:\b\s\w\ir\x\w\llvm_build\bin\clang.exe' -cc1 -internal-isystem 'C:\b\s\w\ir\x\w\llvm_build\lib\clang\19\include' -nostdsysteminc -fsyntax-only -Wno-c++11-narrowing -Wno-literal-conversion -std=c++20 -verify 'C:\b\s\w\ir\x\w\llvm-llvm-project\clang\test\SemaCXX\cxx20-ctad-type-alias.cpp'
# .---command stderr------------
# | error: 'expected-note' diagnostics expected but not seen: 
# |   File C:\b\s\w\ir\x\w\llvm-llvm-project\clang\test\SemaCXX\cxx20-ctad-type-alias.cpp Line 112: cannot deduce template arguments for 'Bar' from 'Foo<int, 4UL>'
# | error: 'expected-note' diagnostics seen but not expected: 
# |   File C:\b\s\w\ir\x\w\llvm-llvm-project\clang\test\SemaCXX\cxx20-ctad-type-alias.cpp Line 112: cannot deduce template arguments for 'Bar' from 'Foo<int, 4ULL>'
# | 2 errors generated.
# `-----------------------------
# error: command failed with exit status: 1

--
```

This happens on our Windows builder. Link to the task: https://ci.chromium.org/ui/p/fuchsia/builders/toolchain.ci/clang-windows-x64/b8747755569988931969/overview

**Comment by mizvekov** [comment]
I already fixed this. We just had to constrain the triple, as there are differences in size_t underlying type which affect diagnostics.

**Comment by hokein** [comment]
Thanks @mizvekov for the quick fix.

**Pull request merged**